server/Cargo.toml

```
@@ -11,26 +11,25 @@ categories.workspace = true
 keywords.workspace = true
 
 [dependencies]
+axum = { version = "0.8.1", features = ["ws"] }
+axum-extra = { version = "0.10.0", features = ["typed-header"] }
 async-trait = "0.1"
 blake2 = "0.10"
-bytes = "1.4"
 futures = "0.3"
-headers = "0.3"
-http = "0.2"
 log = "0.4"
 serde = "1.0"
 serde_json = "1.0"
 subtle = "2.5"
 thiserror = "1.0"
-tokio = { version = "1.25", features = ["sync"] }
-warp =  "0.3"
+tokio = { version = "1.43.0", features = ["sync"] }
+tower-http = { version = "0.6.2", features = ["auth", "cors"] }
 
 nimiq-jsonrpc-core = { workspace = true }
 
 [dev-dependencies]
 anyhow = "1.0"
 pretty_env_logger = "0.5.0"
-tokio = { version = "1.25", features = ["macros", "rt-multi-thread"] }
+tokio = { version = "1.43.0", features = ["macros", "rt-multi-thread"] }
```

I don't think we need to be that strict:
```suggestion
tokio = { version = "1.43", features = ["macros", "rt-multi-thread"] }
```

---

@jsdanielh fixed